src/hooks/useGetPrice.ts

```diff
@@ -1,12 +1,49 @@
 import { useState, useCallback, useEffect } from "react";
 import Decimal from "decimal.js";
+import NodeCache from "node-cache";
 
 import { getOneinchPrice } from "@gnosis.pm/dex-liquidity-provision/scripts/utils/price_utils";
 
 import { TokenDetails } from "types";
+import { ONE_DECIMAL, PRICE_CACHE_TIME } from "utils/constants";
 
 export type PriceSources = "1inch";
 
+const cache = new NodeCache({ stdTTL: PRICE_CACHE_TIME });
+
+/**
+ * Builds 2 cache keys:
+ * 1. Regular for the current token pair
+ * 2. Inverted in case regular is not found.
+ *  We can then invert the price without another query
+ *
+ * @param baseToken
+ * @param quoteToken
+ */
+function buildCacheKeys(
+  baseToken: TokenDetails,
+  quoteToken: TokenDetails
+): [string, string] {
```

Comment:
nit: Since the keys are only ever intended to be used together, and never split, maybe this should return an opaque "keys" type that gets passed into `fetchPriceFromCache` directly?

Reply:
Well, yes, and no.
I need the key exposed afterwards to store the fetched price on the cache.

Reply:
As mentioned above, I'll not use the inverse price anymore, so this won't be a problem.

Reply:
> I need the key exposed afterwards to store the fetched price on the cache.

What I meant was more about how its called, not so much the internals:
```ts
    // Something like this:
    const cacheKeys = buildCacheKeys(baseToken, quoteToken);
    let price: Decimal | null = fetchPriceFromCache(cacheKeys);
    // instead of this:
    const [cacheKey, invertedCacheKey] = buildCacheKeys(baseToken, quoteToken);
    let price: Decimal | null = fetchPriceFromCache(cacheKey, invertedCacheKey);
```

So the outside doesn't have to know about the inner implementation details that each token pair needs a cache key pair.

Reply:
But as you said, this isn't needed anymore.

Reply:
Ah got it. Makes sense indeed!